### ISSUE TITLE
Failed to exec themesconfig script

### ISSUE THREAD

**Issue by dosconcombres** [qgis/qwc2-demo-app#138]
Hello. Trying to install the demo, but on 'yarn start' system gets an error I can't understand. Any advice would be appreciated.

Yarn 1.13.0
Ubuntu 16.04.6 LTS
QWC2App@0.3.0

++++++OUTPUT++++++

0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'run', 'themesconfig' ]
2 info using npm@6.7.0
3 info using node@v11.10.0
4 verbose run-script [ 'prethemesconfig', 'themesconfig', 'postthemesconfig' ]
5 info lifecycle QWC2App@0.3.0~prethemesconfig: QWC2App@0.3.0
6 info lifecycle QWC2App@0.3.0~themesconfig: QWC2App@0.3.0
7 verbose lifecycle QWC2App@0.3.0~themesconfig: unsafe-perm in lifecycle true
8 verbose lifecycle QWC2App@0.3.0~themesconfig: PATH: /usr/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/qwc2-demo-app/node_modules/.bin:/usr/bin:/tmp/yarn--1550985154208-0.21380104265921562:/home/qwc2-demo-app/node_modules/.bin:/usr/local/share/.config/yarn/link/node_modules/.bin:/home/qwc2-demo-app/node_modules/.bin:/usr/local/share/.config/yarn/link/node_modules/.bin:/usr/libexec/lib/node_modules/npm/bin/node-gyp-bin:/usr/lib/node_modules/npm/bin/node-gyp-bin:/usr/bin/node_modules/npm/bin/node-gyp-bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
9 verbose lifecycle QWC2App@0.3.0~themesconfig: CWD: /home/qwc2-demo-app
10 silly lifecycle QWC2App@0.3.0~themesconfig: Args: [ '-c', 'node qwc2/scripts/themesConfig.js' ]
11 silly lifecycle QWC2App@0.3.0~themesconfig: Returned: code: 1  signal: null
12 info lifecycle QWC2App@0.3.0~themesconfig: Failed to exec themesconfig script
13 verbose stack Error: QWC2App@0.3.0 themesconfig: `node qwc2/scripts/themesConfig.js`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:301:16)
13 verbose stack     at EventEmitter.emit (events.js:197:13)
13 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:197:13)
13 verbose stack     at maybeClose (internal/child_process.js:984:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:265:5)
14 verbose pkgid QWC2App@0.3.0
15 verbose cwd /home/qwc2-demo-app
16 verbose Linux 4.4.0-142-generic
17 verbose argv "/usr/bin/node" "/usr/bin/npm" "run" "themesconfig"
18 verbose node v11.10.0
19 verbose npm  v6.7.0
20 error code ELIFECYCLE
21 error errno 1
22 error QWC2App@0.3.0 themesconfig: `node qwc2/scripts/themesConfig.js`
22 error Exit status 1
23 error Failed at the QWC2App@0.3.0 themesconfig script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]

**Comment by manisandro** [comment]
This isn't the actual output on the command line when running `$ yarn start`, is it?

**Comment by dosconcombres** [comment]
Hi @manisandro. Thanks for your reply. No, the prior post comes from the debug log.

Below is the exact output when I run the command. Interestingly I tried installing the app on a clean server and it worked perfectly. It only doesn't work on the server that has QGIS server already installed. Otherwise, the server is basically unmodified. Any advice you might add to walk me through this would be appreciated. Thank you!

$ :/home/qwc2-demo-app# yarn start
yarn run v1.13.0
$ npm run tsupdate && npm run themesconfig && npm run iconfont && webpack-dev-se                                                                                                                                                                                                rver --progress --colors --host 0.0.0.0 --port 8081

> QWC2App@0.3.0 tsupdate /home/qwc2-demo-app
> node qwc2/scripts/updateTranslations.js

Wrote qwc2/translations/data.de-DE
Wrote translations/data.de-DE
Wrote qwc2/translations/data.de-CH
Wrote qwc2/translations/data.en-US
Wrote translations/data.en-US
Wrote qwc2/translations/data.es-ES
Wrote qwc2/translations/data.fr-FR
Wrote translations/data.fr-FR
Wrote qwc2/translations/data.it-IT
Wrote translations/data.it-IT
Wrote qwc2/translations/data.pt-BR
Wrote translations/data.pt-BR
Wrote qwc2/translations/data.ro-RO
Wrote translations/data.ro-RO
Wrote qwc2/translations/data.ru-RU
Wrote translations/data.ru-RU
Wrote qwc2/translations/data.sv-SE
Wrote translations/data.sv-SE
Wrote qwc2/translations/data.cs-CZ

> QWC2App@0.3.0 themesconfig /home/qwc2-demo-app
> node qwc2/scripts/themesConfig.js

/home/qwc2-demo-app/node_modules/deasync/index.js:45
                        throw err
                        ^

Error: getaddrinfo ENOTFOUND ubuntunyc3
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:58:26)
    at Function.module.exports.loopWhile (/home/qwc2-demo-app/node_modules/deasy                                                                                                                                                                                                nc/index.js:71:23)
    at /home/qwc2-demo-app/node_modules/deasync/index.js:41:18
    at Object.<anonymous> (/home/qwc2-demo-app/qwc2/scripts/themesConfig.js:24:4                                                                                                                                                                                                4)
    at Module._compile (internal/modules/cjs/loader.js:738:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:749:10)
    at Module.load (internal/modules/cjs/loader.js:630:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:570:12)
    at Function.Module._load (internal/modules/cjs/loader.js:562:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:801:12)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! QWC2App@0.3.0 themesconfig: `node qwc2/scripts/themesConfig.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the QWC2App@0.3.0 themesconfig script.
npm ERR! This is probably not a problem with npm. There is likely additional log                                                                                                                                                                                                ging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2019-02-26T02_03_12_292Z-debug.log
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this comm                                                                                                                                                                                                and.

**Comment by manisandro** [comment]
> Error: getaddrinfo ENOTFOUND ubuntunyc3

Looks like a hostname lookup failure, so not related to QWC2.

**Comment by dosconcombres** [comment]
Hi @manisandro, Thanks for this invaluable clue. I solved this issue, which is that the hostname of my machine was too long, I guess. I shortened it and now the server starts. I hope this information is helpful to others' who may have run into this problem.

Here are the steps I took... Edit these files to shorten the names of the hosts
`$ cd /etc/hosts`
`$ cd /etc/hostname`
`reboot`
`yarn start` It works!
